### PR TITLE
test(publish): uat of publish command using aws-actions 

### PR DESCRIPTION
**Pull request by saranyailla**
**Issue #, if available:**

**Description of changes:**
- UATs for publish command with and without building the component.  
- Use https://github.com/aws-actions/configure-aws-credentials to manage credentials for session. ToDo: Update this to use other account. This PR is only for testing. 

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
- [ ] Updated the README if applicable
- [ ] Updated or added new unit tests
- [ ] Updated or added new integration tests
- [x] Updated or added new end-to-end tests
- [ ] If your code makes a remote network call, it was tested with a proxy
- [x] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/t_utils.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    s3_client = boto3.client("s3", region_name=region)
+    try:
+        bucket = f"gdk-cli-uat-{region}-{account}"
+        res = s3_client.list_objects(Bucket=bucket)
+        if "Contents" not in res:
+            return
+        for f in res["Contents"]:
+            if f"{component_name}/{component_version}" in f["Key"]:
+                s3_client.delete_object(Bucket=bucket, Key=f["Key"])
+    except Exception as e:
```
Comment (avipinku) [inline uat/t_utils.py:24]: Will this fail the run or report it as passed? Don't want this to go under the radar if it fails to clean up

Comment (saranyailla) [reply uat/t_utils.py:24]: Yes, right now the test fails when the resources are not properly cleaned up.

File: uat/test_uat_publish.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    assert config_file.exists()
+
+    # Update gdk-config file mandatory field like region.
+    with open(str(config_file), "r") as f:
+        config = json.loads(f.read())
+        config["component"][component_name]["author"] = "gdk-cli-uat"
+        config["component"][component_name]["publish"]["region"] = region
+        config["component"][component_name]["publish"]["bucket"] = "gdk-cli-uat"
+    with open(str(config_file), "w") as f:
+        f.write(json.dumps(config))
```
Comment (pallavi03) [inline uat/test_uat_publish.py:30]: can we take this  out as method to reuse?

**Review by avipinku — APPROVED** [review]

File: uat/test_uat_publish.py (hunk 2 above)
Comment (saranyailla) [reply uat/test_uat_publish.py]: Done

**Review by pallavi03 — APPROVED** [review]

**Review by avipinku — APPROVED** [review]

**Pull request merged**